src/pip/_internal/req/req_install.py

```diff
@@ -648,13 +648,13 @@ def uninstall(self, auto_confirm=False, verbose=False):
 
         """
         if not self.check_if_exists():
-            raise UninstallationError(
-                "Cannot uninstall requirement %s, not installed" % (self.name,)
-            )
+            logger.info("Skipping %s as it is not installed." % self.name)
```

Comment:
I think this should be a warning (and I prefer when logging statement split the call between msg and args ^^).
(so `logger.warning("Skipping %s as it is not installed.", self.name))` in this case).

Reply:
Oops!

Nice catch! :)